src/controllers/articles.controller.js

```diff
@@ -0,0 +1,152 @@
+import crypto from 'crypto';
+import slug from 'slug';
+import dotenv from 'dotenv';
+import db from '../database/models';
+import responseGenerator from '../helpers/responseGenerator';
+import BaseRepository from '../repository/base.repository';
+
+dotenv.config();
+
+/**
+ *
+ *
+ * @class ArticleController
+ */
+class ArticleController {
+  /**
+   * @static
+   * @param {object} req - - express request parameter
+   * @param {object} res - - express response parameter
+   * @returns {object} - - article object
+   * @memberof ArticleController
+   */
+  static async createArticle(req, res) {
+    const { id } = req.currentUser;
+    const { title, description, content, image } = req.body;
+
+    try {
+      const article = await BaseRepository.create(db.Article, {
+        title,
+        description,
+        content,
+        image,
+        slug: slug(
+          `${req.body.title}-${crypto.randomBytes(12).toString('base64')}`
+        ).toLowerCase(),
+        authorId: id
+      });
+      return responseGenerator.sendSuccess(res, 201, article);
+    } catch (error) {
+      return responseGenerator.sendError(res, 500, error.message);
+    }
+  }
+
+  /**
+   *
+   *
+   * @static
+   * @param {object} req - - express request parameter
+   * @param {object} res - - express response parameter
+   * @returns {array} - - array of articles
+   * @memberof ArticleController
+   */
+  static async fetchAllArticles(req, res) {
+    try {
+      const articles = await BaseRepository.findAll(db.Article);
+      return responseGenerator.sendSuccess(res, 200, articles);
+    } catch (error) {
+      return responseGenerator.sendError(res, 500, error.message);
+    }
+  }
+
+  /**
+   *
+   *
+   * @static
+   * @param {*} req - - express request parameter
+   * @param {*} res - - express response parameter
+   * @returns {object} - - article object
+   * @memberof ArticleController
+   */
+  static async fetchSpecificArticle(req, res) {
+    const { articleId } = req.params;
+    try {
+      const retrievedArticle = await BaseRepository.findOne(
+        db.Article,
+        articleId
+      );
+      if (retrievedArticle) {
+        return responseGenerator.sendSuccess(res, 200, retrievedArticle);
+      }
+      return responseGenerator.sendError(res, 404, 'Article Not Found');
```

Comment:
Thank you for this review. Can you please be specific on the status code you are suggesting and where exactly it can be used going forward.

Reply:
Hello @tobechukwuobitube, As @tolumide-ng mentioned 404 is not so good in this case. Since the request was actually successful but couldn't return the data you expected. I think you should consider using 400 here. 404 should be used when a route you requested doesn't exist.

Reply:
Thank you @codexempire for your feedback. 
The implementation is scoped on fetching a resource through a `GET` Http method and also was based on research from [this source](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404) and [this source](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200), though there is another similar information from [this research](https://tools.ietf.org/html/rfc7231#section-6.5.4) also.